docs/manuals/source/Appendix/OperatingSystems.rst

```diff
@@ -1,3 +1,114 @@
+
+Hardware Sizing ===============
+
+Database System ---------------
+
+It is always advisable to use the **newest** currently **supported
+|postgresql|** on your platform.  With identical contents, we have observed
+that *|postgresql| 12* required *half of the space* of a *|postgresql| 10*
+Installation.
+
+The by far **biggest table in the bareos database** is the **file** table.  It
+typically consumes about **90-95%** of the database size.
+
+The **size of the file table** depends on the number of files that need to be
+stored and the average size of the filename.
+
+Roughly **1/3 of the file table size** is consumed by the **index**.  To have
+optimal performance, the **RAM size available** for the bareos database should
+be at least the **size of the file table index**.
+
+The database files should be stored on **SSD storage**.
+
+The database configuration should be optimized for the size of memory that the
+database has available.  A good starting point is using pgtune, a website that
+allows to set some basic information and generates a postgresql configuration.
+
+Depending on the number of files and the average length of filenames, the
+**database size** can be **estimated** via the following calculations:
+
+To **calculate the number of files** in the DB, the number of files being
+backed up from all systems needs to be multiplied by the number of times that
+they will be kept in the database.
+
+The **amount of data per file** in the DB, depends on the size of filenames
+that are being backed up, but we have analyzed some real-world examples and
+found that values between **250 and 350 bytes per row** are usual.
+
+*Example:* If **200.000 files** are backed up during a full backup, a full
+backup is run **every week** and the retention of the backups is **1 month**,
+the total amount of files would be
+
+200.000 Files * 4 Full Backups = 800.000 rows in the file table.  Number of Files in
+file table * 300 bytes/row = 240 GB Database size.  About 1/3 of the DB Size
+should be available as RAM, so about 80 GB.
```

Comment:
800.000 rows * 300 bytes/row = 240.000.000 bytes, which is 240 MB not GB.
Maybe we can boil this down to "you will need about 300 MB of space per 1 million files stored".